core/arch/arm/include/sm/extra_smc.h

```diff
@@ -0,0 +1,104 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) STMicroelectronics 2023 - All Rights Reserved
+ */
+#ifndef OPTEE_EXTRA_SMC_H
+#define OPTEE_EXTRA_SMC_H
+
+#include <sm/optee_smc.h>
+
+/*
+ * This file describes the trusted OS additional part.
+ * It exposes the SMC function ID used and the associated functions.
+ * We're following ARM SMC Calling Convention as specified in
+ * http://infocenter.arm.com/help/topic/com.arm.doc.den0028a/index.html
+ *
+ */
+
+#define OPTEE_SMC_OWNER_TRUSTED_OS_EXTRA	U(60)
```

Comment:
Using Trusted OS range makes TF-A BL31 to route the SMC to OP-TEE, whereas a SIP would not reach OP-TEE SPD of TF-A. This is useful when watchdog driver is owned by Trusted OS, not by BL31.

Reply:
The idea is to keep opened this file for other TOS SMC services (if any), not limited to watchdog.
For the range, cf @etienne-lms reply.

Reply:
We have the de facto standard SMC id `0x82003D06` which works perfectly well on v7 systems where I suppose this will be used. If you need another value you can override it in the platform configuration using `CFG_WDT_SM_HANDLER_ID`.

> The idea is to keep opened this file for other TOS SMC services (if any), not limited to watchdog.

My hope is that we'll need none extra at all.

> Using Trusted OS range makes TF-A BL31 to route the SMC to OP-TEE, whereas a SIP would not reach OP-TEE SPD of TF-A. This is useful when watchdog driver is owned by Trusted OS, not by BL31.

Adding random SMC services to OP-TEE is not compatible with FF-A, so in the longer term, it will become a legacy thing. If you're aiming for v8 systems perhaps it's better to try to avoid the legacy track.

Reply:
>We have the de facto standard SMC id 0x82003D06 which works perfectly well on v7 systems where I suppose this will be used. If you need another value you can override it in the platform configuration using CFG_WDT_SM_HANDLER_ID.

You're right, it became standard as put in the Linux driver but the initial target was TF-A SMC call, not TOS implementation. This is only limited to Linux side as OP-TEE doesn't define any HANDLER_ID by default for v7 implementation. Whatever the value, you must define it when WDT is enable.
The first idea was to use a standard ID for Trusted OS as the watchdog is managed in the OP-TEE side and put it as an TOS SMC SIP range.
Going in that way will allow a generic approach for platform using OP-TEE that use the same ID between 32 and 64 bit platform.

>>The idea is to keep opened this file for other TOS SMC services (if any), not limited to watchdog.

> My hope is that we'll need none extra at all.

I hope so..., bet on the future, no issue to rename and only keep it for WDT for now.

> Adding random SMC services to OP-TEE is not compatible with FF-A, so in the longer term, it will become a legacy thing. If you're aiming for v8 systems perhaps it's better to try to avoid the legacy track.

Not agree with you. Platform working on pre armv8.4 will not switch to FF-A (FF-A will not be legacy for such platforms). I'll expect a watchdog ID for FF-A later for those which use it but we need for 64bit platform a way to address the secure watchdog manage by the TOS that will be redirect by the TF-A (current implementation).

Reply:
I don't share your view, but it's pointless to argue about that here. Please make this file exclusive for WDT and we should be done here.